docs/themes/thxvscode/layouts/partials/breadcrumb.html

```diff
@@ -0,0 +1,59 @@
+{{/* 
+  Purpose: To generate breadcrumb navigation for documentation pages. The script processes the current page's URL to extract package names and 
+           section identifiers, mapping them to human-readable names. It constructs a hierarchical breadcrumb trail, indicating the current page's 
+           position within the site's documentation structure.
+  Inputs: The current page's URL and site data (e.g., package names, API details).
+  Output: An HTML paragraph (<p>) containing the breadcrumb links.
+*/}}
+
+{{ $concept := "Packages" }}
+{{ $leafSegment := "" }}
+{{ $leafNodeWithCapitalization := "" }}
+{{ $categoryCapitalized := "" }}
+{{ $leafNodeFound := false }}
+
+{{ $fullURL := trim .Page.RelPermalink "/" }}
+{{ $urlSegments := split $fullURL "/" }}
+{{ $packageName := index $urlSegments 2 }}
```

Comment:
This code will probably break when @chentong7 decides to push her API versioning code in, is there a better way to extract query params other than splitting based on `/`?

Reply:
I'm not sure. I wouldn't be surprised if Hugo offered some built-in macros for doing something like this, but I don't know for sure. 

Let's be sure to at least add a comment explaining our assumptions about the structure we're leaning on here, so devs working here in the future know how to accommodate changes.

Reply:
Found this in the docs `directly accessing URL query parameters in Hugo is not a built-in feature due to its static nature`